src/syntax.ml

```diff
@@ -1067,6 +1153,358 @@ and explicit_list_items :
 
   consume_list_items []
 
+(* Consumes a sequence of table rows which may start with [`Bar].
+
+   This function is called immediately after '{t' ([`Begin_table `Light]) is
+   read. The only "valid" way to exit is by reading a [`Right_brace] token,
+   which is consumed. *)
+and light_table :
+    parent_markup:[< Token.t ] ->
+    parent_markup_location:Loc.span ->
+    input ->
+    Ast.inline_element with_location list list
+    * Ast.alignment list
+    * Ast.inline_element with_location list list list
+    * Loc.span =
+ fun ~parent_markup ~parent_markup_location input ->
+  let valid_align = function
+    | [ { Loc.value = `Word ""; _ } ] -> Some `Default
+    | [ { Loc.value = `Word w; _ } ] ->
+        if String.length w = 1 then
+          match w with "-" -> Some `Default | ":" -> Some `Center | _ -> None
+        else if
+          String.for_all (Char.equal '-') (String.sub w 1 (String.length w - 2))
```

Comment:
This is `String.contains (...) '-'`. Two compat functions can be removed.

This function would be easier to read if it was broken into smaller functions. Also, `valid_align` could benefit from some local variables to avoid copying the string or to store the length of string.

Reply:
But I want to check that it only contains '-'.